test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -270,9 +276,12 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         'await Promise.resolve(); import "node:os"; console.log("executed");',
       ]);
 
-      strictEqual(stderr, '');
-      strictEqual(stdout, 'executed\n');
-      strictEqual(code, 0);
+      match(
+        stderr,
+        /Top-level await is not supported in CommonJS modules\. To use top-level await, add "type": "module" to your package\.json or rename the file to use the \.mjs extension/
```

Comment:
This test is titled “permits top-level `await` above import/export syntax”. Based on the change you made to the assertion, your PR breaks this test.